runtime/lua/vim/filetype.lua

```diff
@@ -863,6 +898,152 @@ local extension = {
+    return "cobol"
+  end,
+  dsl = function(path, bufnr)
+    if getline(bufnr, 1):find("^%s*<!") then
+      return "dsl"
+    end
+    return "structurizr"
+  end,
+  edf = "edif",
+  edfi = "edif",
+  edo = "edif",
+  edn = function(path, bufnr)
+    if getline(bufnr, 1):find("^%s*%(%s*edif>") then
+      return "edif"
+    end
+    return "clojure"
+  end,
+  smil = function(path, bufnr)
+    if getline(bufnr, 1):find("<?%s*xml.*?>") then
+      return "xml"
+    end
+    return "smil"
+  end,
+  smi = function(path, bufnr)
+    if getline(bufnr, 1):find("<smil>") then
+      return "smil"
+    end
+    return "mib"
+  end,
+  install = function(path, bufnr)
+    if getline(bufnr, 1):find("<%?php") then
+      return "php"
+    end
+    vim.fn["dist#ft#SetFileTypeSH"]("bash")
+  end,
+  pm = function(path, bufnr)
+    if getline(bufnr, 1):find("XPM2") then
+      return "xpm2"
+    elseif getline(bufnr, 1):find("XPM") then
+      return "xpm"
+    else
+      return "perl"
+    end
+  end,
+  me = function(path)
+    local filename = vim.fn.fnamemodify(path, ":t")
+    if filename ~= "read.me" and filename ~= "click.me" then
+      return "nroff"
+    end
+  end,
+  reg = function(path, bufnr)
+    local line = getline(bufnr, 1)
+    if line:find("^REGEDIT[0-9]*%s*$") or line:find("^Windows Registry Editor Version %d*%.%d*%s*$") then
+      return "registry"
+    end
+  end,
+  decl = function(path, bufnr)
+    if getline(bufnr, 1, 3):lower():find("^<!sgml") then
+      return "sgmldecl"
+    end
+  end,
+  dcl = function(path, bufnr)
+    if getline(bufnr, 1, 3):lower():find("^<!sgml") then
+      return "sgmldecl"
+    end
+    return "clean"
+  end,
+  dec = function(path, bufnr)
+    if getline(bufnr, 1, 3):lower():find("^<!sgml") then
+      return "sgmldecl"
+    end
+  end,
+  web = function(path, bufnr)
+    if getline(bufnr, 1, 5):find("^%%") then
+      return "web"
+    end
+    return "winbatch"
+  end,
+  ttl = function(path, bufnr)
+    local line = getline(bufnr, 1)
+    if line:find("^@?prefix") or line:find("^@?base") then
+      return "turtle"
+    end
+    return "teraterm"
+  end,
+  am = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ":t"):lower()
+    if not filename:find("makefile.am$") then
+      return "elf"
+    end
+  end,
+  ["m4"] = function(path, bufnr)
+    local filename = vim.fn.fnamemodify(path, ":t"):lower()
+    if not filename:find("html.m4$") and not filename:find("fvwm2rc") then
+      return "m4"
+    end
+  end,
+  hw = function(path, bufnr)
+    if getline(bufnr, 1):find("<%?php") then
+      return "php"
+    end
+    return "virata"
+  end,
+  module = function(path, bufnr)
+    if getline(bufnr, 1):find("<%?php") then
+      return "php"
+    end
+    return "virata"
+  end,
+  pkg = function(path, bufnr)
+    if getline(bufnr, 1):find("<%?php") then
+      return "php"
+    end
+    return "virata"
+  end,
+  ms = function()
+    if not vim.fn["dist#ft#FTnroff"]() then
```

Comment:
What does this function return? (`0` is truthy in Lua, so you usually have to explicitly compare `vim.fn.foo() == 1` across the viml--Lua barrier.)